patches/minecraft/net/minecraft/world/level/block/SugarCaneBlock.java.patch

```diff
@@ -33,6 +33,15 @@
        BlockState blockstate = p_57176_.m_8055_(p_57177_.m_7495_());
        if (blockstate.m_60713_(this)) {
           return true;
+@@ -77,7 +_,7 @@
+             for(Direction direction : Direction.Plane.HORIZONTAL) {
+                BlockState blockstate1 = p_57176_.m_8055_(blockpos.m_142300_(direction));
+                FluidState fluidstate = p_57176_.m_6425_(blockpos.m_142300_(direction));
+-               if (fluidstate.m_76153_(FluidTags.f_13131_) || blockstate1.m_60713_(Blocks.f_50449_)) {
++               if (blockstate1.m_60713_(Blocks.f_50449_) || fluidstate.canHydrate(p_57175_)) {
```

Comment:
This may be fine as is, but I wonder if it should be switched in order to match the order vanilla checks it just so it is a bit easier to see what the patch is doing even though comparing block type is likely less expensive of an operation than some can hydrate implementations may be.